state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -242,27 +245,44 @@ pub struct CcmSwapAmounts {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
-enum CcmSwapState {
-	PrincipalSwapScheduled {
-		principal_swap_id: SwapId,
-		gas_budget: AssetAmount,
-		other_gas_asset: Option<Asset>,
-	},
-	GasSwapScheduled {
-		gas_swap_id: SwapId,
-		principal_output_amount: AssetAmount,
-	},
+enum GasSwapState {
+	OutputReady { gas_budget: AssetAmount },
+	Scheduled { gas_swap_id: SwapId },
+	ToBeScheduled { gas_budget: AssetAmount, other_gas_asset: Asset },
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+struct DCAState {
+	scheduled_chunk_swap_id: Option<SwapId>,
+	remaining_input_amount: AssetAmount,
+	remaining_chunks: u32,
+	accumulated_output_amount: AssetAmount,
+}
```

Comment:
What would this be useful for? Debugging by inspecting storage?

Reply:
I was thinking about things like rpc queries, but yes this too. 

Taking the idea a bit further, storing a history of chunk executions rather than just swap ids would make it easier to reason about correctness, and to simulate dca execution for testing: All of the values in this struct (remaining chunks, amount, total output) can be derived from the initial swap/dca parameters and the history of dca swap executions. Then we only need to mutate a single value (push a new chunk to a vec) rather than relying on consistently mutating multiple values.